tests/invlet_test.cpp

```diff
@@ -46,6 +46,32 @@ enum test_action {
     TEST_ACTION_NUM,
 };
 
+// This is a massive hack that makes this test work without totally rewriting it after #31406
+// The number of null items in item.components is used as a UUID for the purposes of this test
+static void set_uuid( item &it, int uuid )
+{
+    it.components = std::list<item>( static_cast<size_t>( uuid ), item() );
```

Comment:
I believe `item` objects have support for arbitrary properties, which is probably a neater solution than using `components`.  However, I'm not sure how it works.  There's `has_property` and `get_property` functions, but I don't see how you set properties...

Reply:
Properties won't work as they are stored on the itype level not per item instance.  I chose `components` because it's the only thing I could find that wouldn't prevent items with different ids from stacking, which is important for part of this test.

Reply:
Ah yes, makes sense.